### PR TITLE
Adding missing unittest stubs for the platform

### PR DESCRIPTION
**Pull request by Hackwar**
This pull request adds missing unittest stubs to the test suite. Hopefully I caught all that were overlooked at some point or another. It also moves unittests from classes that have been moved to their correct position. (specifically in the database package)

### CODE REVIEW

File: tests/suite/joomla/database/JDatabaseDriverTest.php [hunk 1]
```diff
@@ -0,0 +1,419 @@
+<?php
+
+require_once dirname(__FILE__) . '/../../../../libraries/joomla/database/driver.php';
```
Comment (realityking) [inline tests/suite/joomla/database/JDatabaseDriverTest.php]: Could you change these to be paths off JPATH_PLATFORM. Else this is maintenance heel when we move files around.

**Comment by robschley** [comment]
Aren't we supposed to be trying to get the tests to comply with the coding style guide?

**Comment by Hackwar** [comment]
I thought I already caught all those. Now those should be fixed.

@Rob None of the unittests are in the platform coding style. Should that be changed?

**Comment by LouisLandry** [comment]
Closing this since the unit testing tree has been completely revamped.

**Pull request closed without merging**